Hardware/12-Wifi_Programming_Tutorial.md

```diff
@@ -100,36 +116,59 @@ Hook up the FTDI friend to the Wifi Shield, don't power the Wifi shield through
 
 Now upload using either the Arduino IDE or a custom make solution.
 
-## Sending Data to Wifi Shield
+## Connecting to the Wifi Shield
+
+Be sure that your WiFi Shield is on your local network. Please continue reading if your OpenBCI Wifi Shield is on the same wifi network as your computer.
+
+The steps for connecting to the Wifi Shield and streaming over TCP:
+
+1. Get Wifi Shield On Your Wireless Network
+2. Find IP Address of Wifi Shield
+3. Open a TCP Socket on Host Computer
+4. Send `POST` `/tcp` http request with open socket IP/Port number, can include options for output format (i.e. JSON or RAW output), along with latency.
+5. Send `POST` `/command` http requests for control or for just streaming use GET `/stream/start` or GET `/stream/stop`
+6. Send `POST` `/latency` http requests for tuning, if packets are dropped because older router or poor connection.
+
+The steps for connecting to the Wifi Shield and streaming over MQTT:
+
+1. Get Wifi Shield On Your Wireless Network
+2. Find IP Address of Wifi Shield
+3. Open a TCP Socket on Host Computer
+4. Send `POST` `/mqtt` http request with broker address with optional username and password
+5. Send `POST` `/command` http requests for control
+
+## Get IP Address of Wifi Shield
```

Comment:
There should be a different guide for the things below this line "Programming commands for Shield" or something like that.